pkg/endpoint/endpoint.go

```diff
@@ -620,7 +620,8 @@ func (e *Endpoint) RunK8sCiliumEndpointSync() {
 				// The CEP was not found, this is the first creation of the endpoint
 				cep = &cilium_v2.CiliumEndpoint{
 					ObjectMeta: meta_v1.ObjectMeta{
-						Name: podName,
+						Name:   podName,
+						Labels: k8sMdl.Status.Identity.Labels.GetMap(),
```

Comment:
Something like this should be repeated near line 603. I think we now support changing k8s labels, which means we would have to update the labels on update.